rskj-core/src/main/java/co/rsk/rpc/netty/JsonRpcWeb3ServerHandler.java

```diff
@@ -57,10 +58,18 @@ public JsonRpcWeb3ServerHandler(Web3 service, List<ModuleDescription> filteredMo
     protected void channelRead0(ChannelHandlerContext ctx, ByteBufHolder request) throws Exception {
         ByteBuf responseContent = Unpooled.buffer();
         int responseCode;
+
         try (ByteBufOutputStream os = new ByteBufOutputStream(responseContent);
-             ByteBufInputStream is = new ByteBufInputStream(request.content().retain())){
-            
+             ByteBufInputStream is = new ByteBufInputStream(request.content().retain());
+             ByteBufInputStream isToValidate = new ByteBufInputStream(request.content().duplicate().retain())) {
+
+            validateRequestContent(isToValidate);
```

Comment:
by adding this, request parsing is being done twice, isn't it? what about overriding `JsonRpcBasicServer.handleJsonNodeRequest()` method and validating request id in there?

Reply:
I can try, but the thing is that I need ByteBufInputStream opened. I will check overriding that method or some of those around of it.